### DISCUSSION TITLE
Weighted Moving Average (WMA)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#227] [Indicators]
Community discussion about [Weighted Moving Average (WMA)](https://dotnet.stockindicators.dev/indicators/Wma/#content)

**Comment by Miguelgs25** [comment]
Hi, 
I just discover this repository and I have started to use this indicator. I notice that it is only for close prices. There are some ways to use others? Maybe We could extend the functionality to others (low, high, open prices) by adding a new parameter to the method. What do you think about it?
Thanks.

**Reply by DaveSkender** [reply]
We’ve added that option to SMA and EMA.  It make sense to do that for WMA too.  I’ve added this in v1.22.4

**Reply by DaveSkender** [reply]
And in version 2.0, we've added full chaining options, so you can now do this:

```csharp
var results = quotes
  .Use(CandlePart.High)
  .GetWma(...);
```

**Comment by paramjit2k** [comment]
Is there a way to use MACD: Histogram as a source? 

TradingView options:
![image](https://user-images.githubusercontent.com/15994717/196139540-57daf029-3f2c-4c62-95e0-1442b8290a80.png)

**Reply by DaveSkender** [reply]
Our chaining model for [MACD](https://dotnet.stockindicators.dev/indicators/Macd/#chaining) currently uses the `Macd` value, so normally you’d simple do this:

```csharp
var results = quotes
  .GetMacd(..)
  .GetWma(..);
```

However, since you want to use a different value, you’ll have to map differently.  Try this:

```csharp
var results = quotes
  .GetMacd(..)
  .Select(x => (x.Date, x.Histogram.NullToNaN()))
  .GetWma(..);
```

**Reply by DaveSkender** [reply]
Just an FYI, as a side note, when choosing the the default chaining value `Macd` I remember having this conversation on which value is more important.  I can be convinced to switch this to `Histogram`, but am not sure yet which is more common in the chaining use case.  I can only pick one and we chose `Macd`.

**Reply by DaveSkender** [reply]
I considered a design where you can simply type the property, but wasn’t happy with the performance since it [relied on slow Reflection methods](https://stackoverflow.com/questions/13292078/refer-to-a-property-name-by-variable).  That’s why we use a default value for chaning.

**Reply by paramjit2k** [reply]
I tried this already:

var results = quotes
  .GetMacd(..)
  .GetWma(..);

But it is not giving similar results.

I will try to use the next one as suggested by you.
